### ISSUE TITLE
Cuesheet support

### ISSUE THREAD

**Issue by serrq** [OxygenCobalt/Auxio#762]
Recently I downloaded full concert (approximately 2 h 30 min) in one only tracks.

I noticed that moving the playing cursor isn't precise and very difficult to skip between songs in a same tracks.

So I ask enabling bookmarks (creation and their use) during the track's playing.

**Comment by OxygenCobalt** [comment]
This sounds like cuesheets is basically what you want. They are files that outline a way of splitting up a huge audio file into several song entries. 

I've rejected this earlier since it was basically impossible in Auxio's architecture, but ever since I added gapless playback support and assuming #322, this actually could become feasible.

**Comment by OxygenCobalt** [comment]
Yes, cuesheets are a file that points towards one huge audio file (of any time) followed a bunch of timestamp ranges that act as the "Songs".

Auxio would read the cuesheet, find the corresponding audio file, then create "Songs" based on those entries that would in practice open up the huge audio file and just seek to the specific region specified in the cuesheet file. You could then play that huge audio file as if it was just an album.

**Comment by OxygenCobalt** [comment]
That's part of the cuesheet file. Auxio does not support them yet though, so you will have to wait.

Here's a wikipedia article: https://en.wikipedia.org/wiki/Cue_sheet_(computing)

For example, here's how you'd specify the timestamps in a real cuesheet file.

```
REM GENRE [Electronica](https://en.wikipedia.org/wiki/Electronica)
REM DATE 1998
PERFORMER "Faithless"
TITLE "Live in Berlin"
FILE "Faithless - Live in Berlin.mp3" MP3
  TRACK 01 AUDIO
    TITLE "Reverence"
    PERFORMER "Faithless"
    INDEX 01 00:00:00
  TRACK 02 AUDIO
    TITLE "She's My Baby"
    PERFORMER "Faithless"
    INDEX 01 06:42:
```

This would specify a track called "Reverence" that starts at 0:00 and ends at 6:42, and then another track called "She's My Baby" that starts at 6:42 and continues until the end of the audio file.

**Comment by OxygenCobalt** [comment]
A button like that would:

1. Not be useful to most users. Most people who listen to long concerts either split their files manually in an audio editor or use cuesheets. I believe there are programs that generate cuesheet files easily as well.
2. Would clutter the UI to an extreme extent. The playback UI has basically no semantic space to add any more elements. This is also a huge reason why I don't add things like lyrics, there's just nowhere to put it.

In general, I would rather work on a more general feature that accomplishes this use-case among many others rather than a very narrow feature that only accomplishes what you specifically want at the cost of a bunch clunkier and confusing experience.
